app/src/app/views/admin-page/user-accounts-page/user-accounts-page.component.ts

```diff
@@ -1,23 +1,109 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnDestroy, OnInit, ViewChild, ViewEncapsulation } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
 import { UserAccount } from 'src/app/classes/authn/user-account';
 import { Paginated, RestApiConnectorService } from 'src/app/services/connectors/rest-api/rest-api-connector.service';
+import { Role } from '../../../classes/authn/role';
+import { MatPaginator } from '@angular/material/paginator';
+import { AuthenticationService } from '../../../services/connectors/authentication/authentication.service';
+import { Status } from '../../../classes/authn/status';
 
 @Component({
     selector: 'app-user-accounts-page',
     templateUrl: './user-accounts-page.component.html',
-    styleUrls: ['./user-accounts-page.component.scss']
+    styleUrls: ['./user-accounts-page.component.scss'],
+    encapsulation: ViewEncapsulation.None
 })
 export class UserAccountsPageComponent implements OnInit {
+    @ViewChild(MatPaginator) paginator: MatPaginator;
     public userAccounts$: Observable<Paginated<UserAccount>>;
+    public userAccounts: UserAccount[];
+    public filteredAccounts: UserAccount[];
+    public columnsToDisplay: string[];
+    public filterOptions: any[];
+    public totalObjectCount: number = 0;
+    public userSubscription: Subscription;
+    public selectedFilters: string[];
+    public get currentUser(): UserAccount { return this.authenticationService.currentUser; }
+    public get isAdmin(): boolean { return this.authenticationService.isAuthorized([Role.Admin]); }
 
-    constructor(private restAPIConnector: RestApiConnectorService) { }
+    constructor(private restAPIConnector: RestApiConnectorService, private authenticationService: AuthenticationService) {
+        this.columnsToDisplay = ['username', 'email', 'status', 'role'];
+        this.filterOptions = [
+            {
+                name: 'Status',
+                values: Object.values(Status as {})
+            },
+            {
+                name: 'Role',
+                values: Object.values(Role as {}).filter((r) => r !== 'none')
```

Comment:
An admin should be able to set a user's role to `none`. This would ideally be used for newly registered users & for inactive users.